packages/vm/src/runTx.ts

```diff
@@ -247,27 +248,56 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
     }
   }
 
+  // Check balance against max potential cost (for EIP 1559 and 4844)
+  let maxCost = tx.value
+  let dataGasPrice = BigInt(0)
+  let totalDataGas = BigInt(0)
   if (tx.supports(Capability.EIP1559FeeMarket)) {
     // EIP-1559 spec:
     // The signer must be able to afford the transaction
     // `assert balance >= gas_limit * max_fee_per_gas`
-    const cost = tx.gasLimit * (tx as FeeMarketEIP1559Transaction).maxFeePerGas + tx.value
-    if (balance < cost) {
-      if (opts.skipBalance === true && fromAccount.balance < cost) {
-        // if skipBalance, ensure caller balance is enough to run transaction
-        fromAccount.balance = cost
-        await this.stateManager.putAccount(caller, fromAccount)
-      } else {
-        const msg = _errorMsg(
-          `sender doesn't have enough funds to send tx. The max cost is: ${cost} and the sender's account (${caller}) only has: ${balance}`,
-          this,
-          block,
-          tx
-        )
-        throw new Error(msg)
-      }
+    maxCost += tx.gasLimit * (tx as FeeMarketEIP1559Transaction).maxFeePerGas
+  }
+
+  if (this._common.isActivatedEIP(4844) && tx instanceof BlobEIP4844Transaction) {
+    // EIP-4844 spec
+    // the signer must be able to afford the transaction
+    // assert signer(tx).balance >= tx.message.gas * tx.message.max_fee_per_gas + get_total_data_gas(tx) * tx.message.max_fee_per_data_gas
+    const castTx = tx as BlobEIP4844Transaction
+    totalDataGas =
+      castTx.common.param('gasConfig', 'dataGasPerBlob') * BigInt(castTx.versionedHashes.length)
```

Comment:
I wonder if this reoccuring code part `castTx.versionedHashes.length` deserves a dedicated method/alias `Transaction.numDataBlobs()` or the like. 🤔 Guess that would make things more readable.

Reply:
`totalDataGas = castTx.common.param('gasConfig', 'dataGasPerBlob') * BigInt(castTx.numBlobs())`

Ah yeah. Nice! 🙂